spec/factories.rb

```diff
@@ -6,21 +6,41 @@
   end
 
   factory :category do
-    association(:post)
-
     sequence(:name) { |n| "Category #{n}" }
+
+    trait :post_with_categories do
+      transient do
+        categories_count { 1 }
+      end
+
+      after(:create) do |post, evaluator|
+        create_list(:category, evaluator.categories_count, posts: [post])
```

Comment:
This trait is nested within the category factory, which I think means that `post` here is actually referring to a category. That might be why we had trouble getting this to work.

Reply:
Oh no wait, I think I remember you moved this here to get it out of the way!